### ISSUE TITLE
Handling of multiple responses

### ISSUE THREAD

**Issue by wpt-issue-mover** [w3c/web-platform-tests#8374]
Originally posted as https://github.com/w3c/wptserve/issues/117 by @annevk on 05 Apr 2017, 08:11 UTC:

> If we want to test 1xx responses, we need to be able to respond with one, and then follow up with another or a final non-1xx response.

**Comment by foolip** [comment]
@annevk, can you elaborate on what's needed here? Which tests can't be written because this is missing?

**Comment by annevk** [comment]
Basically ensuring that everything that does a fetch does not do bad things if the scenario in OP happens. But also testing https://tools.ietf.org/html/draft-ietf-httpbis-early-hints I suppose if browsers implement that (Chrome might).

**Comment by foolip** [comment]
Would that require some changes to wptserve? I think there's already a way to send a raw response and to save state across requests.

**Comment by annevk** [comment]
@foolip I'm not sure why you'd need to save state here? I think I tried raw responses, but it's been ages.

**Comment by foolip** [comment]
If raw responses are all that's needed, then probably this already works then. Can you change the labels to the specs that need this, and send back to infra if it doesn't work?

**Comment by fred-wang** [comment]
I'm investigating [fetchpriority in Link headers](https://html.spec.whatwg.org/multipage/semantics.html#link-options-fetch-priority) for [early hints](https://html.spec.whatwg.org/multipage/semantics.html#early-hints) in [Mozilla's bug 1882084](https://bugzilla.mozilla.org/show_bug.cgi?id=1882084) and also could not find in the [wpt doc](https://web-platform-tests.org/writing-tests/server-features.html) how to send multiple headers (more specifically 1xx informational response, followed by 2xx successful response).

@foolip: What do you mean by raw responses? Using `.asis` files?

**Comment by fred-wang** [comment]
I just found https://github.com/web-platform-tests/wpt/commits/master/loading/early-hints which has tests for early hints.

**Comment by fred-wang** [comment]
@foolip Right, it seems they are just using `response.writer.write_raw_header_frame` to write a raw 103 response in `early-hints-test-loader.h2.py` so I can do the same with fetchpriority (I had started https://phabricator.services.mozilla.com/D209632 but have been focusing on something else so haven't really tried yet).
